diff -ru tuw_cvodeevolve/tuw_cvodeevolve.cc tuw_cvodeevolve-new/tuw_cvodeevolve.cc
--- tuw_cvodeevolve/tuw_cvodeevolve.cc	2022-01-22 02:39:23.919149032 -0500
+++ tuw_cvodeevolve-new/tuw_cvodeevolve.cc	2022-01-22 01:45:32.000000000 -0500
@@ -4,7 +4,7 @@
  *
  */
 
-#include <float.h>
+#include <cfloat>
 
 #include "nb.h"
 
diff -ru tuw_cvodeevolve/tuw_cvodeevolve.h tuw_cvodeevolve-new/tuw_cvodeevolve.h
--- tuw_cvodeevolve/tuw_cvodeevolve.h	2022-01-22 02:39:23.925149259 -0500
+++ tuw_cvodeevolve-new/tuw_cvodeevolve.h	2022-01-22 01:45:32.000000000 -0500
@@ -7,7 +7,7 @@
 #ifndef _TUW_CVODEEVOLVE
 #define _TUW_CVODEEVOLVE
 
-#include <stdio.h>  // for FILE* logfile.
+#include <cstdio>  // for FILE* logfile.
 
 #include "director.h"
 #include "ext.h"
