Only in xf_thermheunevolve: versdate.txt
diff -ru xf_thermheunevolve/xf_thermheunevolve.cc xf_thermheunevolve-new/xf_thermheunevolve.cc
--- xf_thermheunevolve/xf_thermheunevolve.cc	2022-01-22 02:39:23.927149334 -0500
+++ xf_thermheunevolve-new/xf_thermheunevolve.cc	2022-01-22 02:31:37.000000000 -0500
@@ -11,10 +11,10 @@
  *
  */
 
-#include <float.h>
+#include <cfloat>
 #include <string>
-#include <stdlib.h>
-#include <time.h>
+#include <cstdlib>
+#include <ctime>
 
 #include "nb.h"
 #include "director.h"
@@ -75,8 +75,8 @@
     max_step_decrease(0.03125), max_step_increase_limit(4.0),
     max_step_increase_adj_ratio(1.9),
     reject_goal(0.05), reject_ratio(0.05),
-    KBoltzmann(1.38062e-23),tmpStepsize(1e-17),
-    has_tempscript(0),temperature(0.),
+    KBoltzmann(1.38062e-23),temperature(0.),
+    tmpStepsize(1e-17),has_tempscript(0),
     energy_state_id(0),next_timestep(1e-27),
     rkstep_ptr(NULL)
 {
@@ -337,7 +337,7 @@
   }
 
   const OC_INDEX size = mesh->Size();
-  OC_INDEX i,j;
+  OC_INDEX i;
 
   alpha_init->FillMeshValue(mesh,alpha);
   gamma_init->FillMeshValue(mesh,gamma);
@@ -1028,7 +1028,7 @@
   GetEnergyDensity(next_state_key.GetReadReference(),temp_energy,
                    &vtmpA,NULL,pE_pt);
   UpdateHfluct(cstate->mesh);// Update the random field before calculating slopes
-  Oxs_SimState& workstate = next_state_key.GetWriteReference();
+  // Oxs_SimState& workstate = next_state_key.GetWriteReference();
   tmpStepsize = stepsize;
   Calculate_dm_dt(next_state_key.GetReadReference(),vtmpA,pE_pt,
                   vtmpA,max_dm_dt,dE_dt,timestep_lower_bound);
